runtime/particle-spec.js

```diff
@@ -112,17 +113,29 @@ class ParticleSpec {
   toLiteral() {
     let {args, name, verbs, transient, description, implFile, affordance, slots} = this._model;
     args = args.map(a => {
-      let {type, direction, name} = a;
+      let {type, direction, name, isOptional} = a;
       type = type.toLiteral();
-      return {type, direction, name};
+      return {type, direction, name, isOptional};
     });
     return {args, name, verbs, transient, description, implFile, affordance, slots};
   }
 
-  static fromLiteral(literal) {
+  static fromLiteral(literal, typeVarMap) {
     let {args, name, verbs, transient, description, implFile, affordance, slots} = literal;
-    args = args.map(({type, direction, name}) => ({type: Type.fromLiteral(type), direction, name}));
-    return new ParticleSpec({args, name, verbs, transient, description, implFile, affordance, slots});
+    args = args.map(({type, direction, name, isOptional}) => ({type: Type.fromLiteral(type), direction, name, isOptional}));
+    return new ParticleSpec({args, name, verbs, transient, description, implFile, affordance, slots}, typeVarMap);
+  }
+
+  clone() {
+    let typeVarMapClone = new Map();
```

Comment:
I'm a bit stuck on this bit, are you sure we need to clone the type var map? It seems to me that constructing the particle spec, constructs connection specs which will then repopulate the map.

Reply:
I got rid of it for now. 
Instead, i added cloning of the hosted particle recipe in manifest parsing.
I'll merge it for now, and try to make progress with the other PR (the type resolution) and see if the errors go away.